File: packages/backend/src/main/java/com/bookspank/backend/service/ReadingListService.java
package com.bookspank.backend.service;

import java.util.List;

import com.bookspank.backend.dto.PostBookRequest;
import com.bookspank.backend.model.ReadingListEntry;

public interface ReadingListService {

    public List<ReadingListEntry> getReadingList(Integer userId, Integer limit, Integer offset);

    public void addBook(Integer userId, PostBookRequest form);

    public void removeBook(Integer userId, Integer bookId);

    public void updateWantToReadNext(Integer userId, Integer bookId, Boolean wantToReadNext);
}

File: packages/aws-nextjs/lib/service/client/ReadingListService.ts
import { OpenLibrarySearchResponse } from "@/dto/OpenLibrarySearchResponse";
import { ReadingListEntry } from "@/model/ReadingListEntry";

export interface IReadingListService {

    /**
     * Assuming currently all books are added to 
     * @param userId 
     * @param bookInfo 
     */
    addBook(userId: number, bookInfo: OpenLibrarySearchResponse): Promise<void>;

    removeBook(userId: number, bookId: number): Promise<void>;

    getReadingList(userId: number, limit: number, offset: number): Promise<ReadingListEntry[]>;

    /**
     * Assuming all books below will keep same position, all above new position will decrease? position by one
     * @param userId 
     * @param bookId 
     * @param newPosition 
     */
    updateBookPosition(userId: number, bookId: number, origPosition: number, newPosition: number): Promise<void>;

}